packages/gatsby/src/utils/merge-gatsby-config.ts

```diff
@@ -1,30 +1,16 @@
-const _ = require(`lodash`)
-/**
- * Defines how a theme object is merged with the user's config
- */
-module.exports = (a, b) => {
-  // a and b are gatsby configs, If they have keys, that means there are values to merge
-  const allGatsbyConfigKeysWithAValue = _.uniq(
-    Object.keys(a).concat(Object.keys(b))
-  )
+import _ from "lodash"
+import { IGatsbyConfig } from "../redux/types"
 
-  // reduce the array of mergable keys into a single gatsby config object
-  const mergedConfig = allGatsbyConfigKeysWithAValue.reduce(
-    (config, gatsbyConfigKey) => {
-      // choose a merge function for the config key if there's one defined,
-      // otherwise use the default value merge function
-      const mergeFn = howToMerge[gatsbyConfigKey] || howToMerge.byDefault
-      return {
-        ...config,
-        [gatsbyConfigKey]: mergeFn(a[gatsbyConfigKey], b[gatsbyConfigKey]),
-      }
-    },
-    {}
-  )
-
-  // return the fully merged config
-  return mergedConfig
-}
+type ConfigKey = keyof IGatsbyConfig
+type Metadata = IGatsbyConfig["siteMetadata"]
+type Plugins = IGatsbyConfig["plugins"]
+type PluginEntry =
+  | string
+  | {
+      resolve: string
+      options: Record<string, unknown>
```

Comment:
```suggestion
      options?: Record<string, unknown>
```

`options` are ... optional :) (ba-dum-tsss)